contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/preplanned/optimizer/PreplannedDrtOptimizer.java

```diff
@@ -203,7 +201,7 @@ public record PreplannedSchedules(Map<PreplannedRequestKey, Id<DvrpVehicle>> pre
 									  Map<PreplannedRequestKey, PreplannedRequest> unassignedRequests) {
 	}
 
-	public record PreplannedRequestKey(Id<Person> passengerId, Id<Link> fromLinkId, Id<Link> toLinkId) {
+	public record PreplannedRequestKey(List<Id<Person>> passengerIds, Id<Link> fromLinkId, Id<Link> toLinkId) {
```

Comment:
Here, it may be better to have a set instead of a list, because two "seemingly identical" keys, but with different order of passengerIds will not be identical.